app/grandchallenge/cases/views.py

```diff
@@ -144,9 +144,13 @@ def get(self, request, *args, **kwargs):
                 },
             )
             return HttpResponse(html_content)
-        elif current_value and (
-            Image.objects.filter(pk=current_value).exists()
-            or UserUpload.objects.filter(pk=current_value).exists()
+        elif (
+            widget_name == WidgetChoices.IMAGE_SELECTED.name
```

Comment:
This part of the condition will work anyway without checking for `widget_name` value, but it is more robust to check for it.